### PR TITLE
Add new field for vulnerability evaluation status

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#25482|

### PR Summary: 

This pull request introduces a new field, `vulnerability.under_evaluation`, in the ECS data structure. The purpose of this field is to indicate whether a vulnerability is still under evaluation or disputed. This flag is relevant when the vulnerability status is **Awaiting analysis (NVD)**, and no conclusive data is available from the ADP.

#### Key Changes:
- **New Field**: `vulnerability.under_evaluation`
  - The field is set to `true` if the following conditions are not met:
    - A valid base score is present (`scoreBase()` is non-zero).
    - The vulnerability has non-empty classification and severity information.
  - Example logic:

```cpp
const auto isUnderEvaluation = [&returnData]()
{
    return Utils::floatToDoubleRound(returnData.data->scoreBase(), 2) == 0 ||
           returnData.data->classification()->str().empty() ||
           returnData.data->severity()->str().empty();
};
```

#### Condition:
The field will be `true` when:
1. The base score is either missing or zero.
2. The classification or severity data is empty.

This additional field improves the ability to track the evaluation status of vulnerabilities, especially when a vulnerability is awaiting the analysis from the NVD and no further classification or severity details are provided by ADP.

---------------------------

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp [hunk 1]
```diff
@@ -229,6 +229,9 @@ class TEventDetailsBuilder final : public AbstractHandler<std::shared_ptr<TScanC
                 auto vulnerabilityDetection = PolicyManager::instance().getVulnerabilityDetection();
                 ecsData["wazuh"]["cluster"]["name"] = data->clusterName();
                 ecsData["wazuh"]["schema"]["version"] = WAZUH_SCHEMA_VERSION;
+                ecsData["wazuh"]["vulnerability"]["under_evaluation"] =
+                    !(Utils::floatToDoubleRound(returnData.data->scoreBase(), 2) &&
+                      !returnData.data->classification()->str().empty() && !returnData.data->severity()->str().empty());
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp:234]: Improve (complexity): I would move this to a lambda or a private method to reduce cognitive complexity

```c++
                const auto isUnderEvaluation = [&returnData]()
                {
                    if (Utils::floatToDoubleRound(returnData.data->scoreBase(), 2) == 0)
                    {
                        return false;
                    }

                    if (returnData.data->classification()->str().empty())
                    {
                        return false;
                    }

                    if (returnData.data->severity()->str().empty())
                    {
                        return false;
                    }

                    return true;
                };

                ecsData["wazuh"]["vulnerability"]["under_evaluation"] = isUnderEvaluation();
                ```

File: src/wazuh_modules/vulnerability_scanner/tests/unit/eventDetailsBuilder_test.cpp [hunk 2]
```diff
@@ -809,3 +809,371 @@ TEST_F(EventDetailsBuilderTest, TestSuccessfulOsInserted)
...
+    auto clusterName = PolicyManager::instance().getClusterName();
+    EXPECT_STREQ(elementData.at("wazuh").at("cluster").at("name").get_ref<const std::string&>().c_str(),
+                 clusterName.c_str());
+    EXPECT_STREQ(elementData.at("wazuh").at("schema").at("version").get_ref<const std::string&>().c_str(),
+                 WAZUH_SCHEMA_VERSION);
+    EXPECT_STREQ(elementData.at("vulnerability").at("published_at").get_ref<const std::string&>().c_str(),
+                 GetVulnerabilityDescription(fbBuilder.GetBufferPointer())->datePublished()->c_str());
+    EXPECT_TRUE(elementData.at("vulnerability").at("detected_at").get_ref<const std::string&>() <=
+                Utils::getCurrentISO8601());
+}
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/tests/unit/eventDetailsBuilder_test.cpp:1179]: Add missing new line
```suggestion
}

```

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp (hunk 1 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp]: Thank you! Nice catch! Just a quick note on why we refactored the `isUnderEvaluation` function in our last commit in a different way

We streamlined the logic into a single return statement, making it much easier to read and maintain. This also cuts down on repetitive `if` checks and improves efficiency with short-circuit evaluation. On top of that, it’ll be simpler to update if we need to add more conditions in the future.

File: src/wazuh_modules/vulnerability_scanner/tests/unit/eventDetailsBuilder_test.cpp (hunk 2 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/tests/unit/eventDetailsBuilder_test.cpp]: Fixed!

File: src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json [hunk 3]
```diff
@@ -239,10 +239,18 @@
...
+                }
+              }
             }
           }
         }
       }
     }
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json]: Fixed!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp (hunk 1 above)
Comment (sebasfalcone) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp]: I liked the mix between both ideas!

File: src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json [hunk 4]
```diff
@@ -239,6 +239,14 @@
...
                 }
               }
+            },
+            "vulnerability": {
+              "properties": {
+                "under_evaluation": {
+                  "ignore_above": 1024,
+                  "type": "bool"
+                }
+              }
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json:249]: Requirements have changed:
- https://github.com/wazuh/wazuh/issues/25482#issuecomment-2333840078

Please update the PR description as well

**Review by sebasfalcone — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json (hunk 4 above)
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json]: Done!

**Review by sebasfalcone — APPROVED** [review]

**Pull request merged**